tendrl/gluster_integration/objects/volume/atoms/validate_shrink_volume_inputs/__init__.py

```diff
@@ -0,0 +1,221 @@
+import etcd
+import json
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons import objects
+from tendrl.gluster_integration.objects.volume import Volume
+
+
+class ValidateShrinkVolumeInputs(objects.BaseAtom):
+    obj = Volume
+    def __init__(self, *args, **kwargs):
+        super(ValidateShrinkVolumeInputs, self).__init__(*args, **kwargs)
+
+    def run(self):
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Checking if inputs for shrink vol are valid for %s" %
+                    self.parameters['Volume.volname']
+                },
+                job_id=self.parameters["job_id"],
+                flow_id=self.parameters["flow_id"],
+                cluster_id=NS.tendrl_context.integration_id,
+            )
+        )
+        
+        try:
+            fetched_volume = Volume(
+                vol_id=self.parameters['Volume.vol_id']
+            ).load()
+        except etcd.EtcdKetNotFound:
+            Event(
+                Message(
+                    priority="info",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Volume %s does not exist" %
+                        self.parameters['Volume.volname']
+                    },
+                    job_id=self.parameters["job_id"],
+                    flow_id=self.parameters["flow_id"],
+                    cluster_id=NS.tendrl_context.integration_id,
+                )
+            )
+            return False
+        try:
+            result = NS.etcd_orm.client.read(
+                "clusters/%s/Volumes/%s/Bricks" % (
+                    NS.tendrl_context.integration_id,
+                    self.parameters['Volume.vol_id']
+                ),
+                sorted=True
```

Comment:
sorting is not needed with latest change. With latest code we are using sequence number and am using that below to get the sub volumes.